### DISCUSSION TITLE
Unknown variable with "defpoll"

### DISCUSSION THREAD

**Discussion by BigfootN** [elkowar/eww#347] [Q&A]
Hello,

I don't understand what I am doing wrong. So here is the problem.

I created a `defpoll` variable:

```
(defpoll active_workspace :interval "1s"
     "scripts/active_workspace")
```

I then use the variable `active_workspace` to display it in a simple `box`.

It seems to work, I can see well the active workspace. *But*, I see, with the `eww logs` :

```
error: Unknown variable active_workspace
...
Hint: If you meant to use the literal value "active_workspace", surround the value in quotes
```

So I would like to know who is right and who is wrong since it seems to work and yet it seems it indicate an error.

Thank you very much for any help

**Comment by druskus20** [comment]
Could you post your entire config and eww-logs?

**Comment by BigfootN** [comment]
Thank you for your quick response and please excuse me for my late one.

So after doing some digging and testing I think I might have found a solution with the explanation. Despite having read several time the documentation, with the example, it still doesn't seem really clear to me.

So, like I said before I created a `defpoll` variable:

```lisp
(defpoll active_workspace :interval "1s"
     "scripts/active_workspace")
```

But I thought that by using it this way, for example:

```lisp
(box :class active_workspace) ; just for the example
```
would work.

But I realized that I had to use it this way:

```lisp
(box :class "${active_workspace}") ; again just for the example
```

Then I `eww logs` didn't seem to report any errors.

Maybe I haven't read the documentation thoroughly enough, but it doesn't seem really clear in the documentation on how to use those type of variables and when to use `{...}` or `"${...}"`

Thank you very much for your help hoping I was clear enough

**Reply by druskus20** [reply]
Glad you have figured it out, we probably should improve error msg's for that
